presto-native-execution/presto_cpp/main/types/FunctionMetadata.cpp

```diff
@@ -134,6 +134,25 @@ const std::vector<protocol::TypeVariableConstraint> getTypeVariableConstraints(
   return typeVariableConstraints;
 }
 
+const std::vector<protocol::LongVariableConstraint> getLongVariableConstraints(
+    const FunctionSignature& functionSignature) {
+  std::vector<protocol::LongVariableConstraint> longVariableConstraints;
+  const auto functionVariables = functionSignature.variables();
+  for (const auto& [name, signature] : functionVariables) {
+    if (signature.isIntegerParameter()) {
+      protocol::LongVariableConstraint longVariableConstraint;
+      longVariableConstraint.name =
+          boost::algorithm::to_lower_copy(signature.name());
+      longVariableConstraint.expression =
+          boost::algorithm::to_lower_copy(signature.constraint());
+      if (!longVariableConstraint.expression.empty()) {
```

Comment:
Can we avoid delaying this check so far along ? In line 142 itself you can check if signature.constraint() is not empty and avoid creating LongVariableConstraint entirely.